test/e2e/usersignup_test.go

```diff
@@ -528,6 +532,73 @@ func (s *userSignupIntegrationTest) TestTargetClusterSelectedAutomatically() {
 	VerifyResourcesProvisionedForSignup(s.T(), s.Awaitilities, userSignup, "deactivate30", "base")
 }
 
+func (s *userSignupIntegrationTest) TestTransformUsernameWithSpaceConflict() {
+	// given
+	conflictingSpace, _, _ := CreateSpace(s.T(), s.Awaitilities, testcommonspace.WithName("conflicting"))
+
+	// when
+	userSignup, _ := NewSignupRequest(s.Awaitilities).
+		Username(conflictingSpace.Name).
+		TargetCluster(s.Member1()).
+		ManuallyApprove().
+		EnsureMUR().
+		RequireConditions(wait.ConditionSet(wait.Default(), wait.ApprovedByAdmin())...).
+		Execute(s.T()).Resources()
+
+	// then
+	compliantUsername := userSignup.Status.CompliantUsername
+	require.Equal(s.T(), fmt.Sprintf("%s-2", conflictingSpace.Name), compliantUsername)
```

Comment:
I'm a bit lost: why is the usersignup's `CompliantUserName` already set to something like `conflicting-2` at the beginning of the test?